async_service/base.py

```diff
@@ -326,18 +326,19 @@ async def _run_and_manage_task(self, task: TaskAPI) -> None:
                 if self.is_cancelled:
                     pass
                 else:
-                    if isinstance(task, TaskWithChildrenAPI):
-                        new_parent = task.parent
-                        for child in task.children:
-                            child.parent = new_parent
-                            self._add_child_task(new_parent, child)
-                            self.logger.debug(
-                                "Daemon %s left child task (%s) behind, reassigning it to %s",
-                                task,
-                                child,
-                                new_parent or "root",
-                            )
                     raise
+            finally:
+                if isinstance(task, TaskWithChildrenAPI):
```

Comment:
So this now runs on any exception, but it also runs if hde `DaemonTaskExit` is raised and `self.is_cancelled` is `True`. It seems like that was intentionally skipped before. Do we want to add a `not self.is_cancelled` check here, too? (Or are you saying it was a bug to skip the reassignment on service cancellation?)

Reply:
I think it was a bug. I can't think of a reason why we wouldn't want to re-assign child tasks when their parent exit as that means they won't be cancelled by `_real_handle_cancelled()` and cause a LifecycleError